src/test/regress/sql/join.sql

```diff
@@ -180,6 +180,13 @@ SELECT * FROM abcd first join abcd second USING(b) join abcd third on first.b=th
 END;
 
 
+-- test join with geqo
+CREATE TABLE dist (a int, b int);
+SELECT create_distributed_table('dist', 'a');
+INSERT INTO dist VALUES (1, 1), (2, 2), (3, 3);
 
+SET geqo_threshold to 2;
+
+SELECT count(*) FROM dist d1 LEFT JOIN dist d2 ON d1.a = d2.a LEFT JOIN dist d3 ON d3.a = d2.a;
```

Comment:
can we add few more tests to be defensive for future changes? I've not tried running the tests, so please make sure that the comment matches the query type

```SQL
-- JOINs  with CTEs:
WITH cte_1 AS (SELECT  * FROM dist OFFSET 0)  SELECT count(*) FROM dist d1 LEFT JOIN dist d2 ON d1.a = d2.a LEFT JOIN dist d3 ON d3.a = d2.a LEFT JOIN cte_1 ON true;
WITH cte_1 AS (SELECT  * FROM dist OFFSET 0),     
     cte_2 AS (SELECT * FROM dist OFFSET 0),
     cte_3 AS (SELECT * FROM dist OFFSET 0)
 SELECT count(*) FROM cte_1 d1 LEFT JOIN cte_2 d2 ON d1.a = d2.a LEFT JOIN cte_3 d3 ON d3.a = d2.a LEFT JOIN cte_1 ON true;
                                                                               


-- Inner JOIN:
 SELECT count(*) FROM dist d1 JOIN dist d2 ON d1.a = d2.a JOIN dist d3 ON d3.a = d2.a;

-- subquery join
 SELECT count(*) FROM (SELECT *,random() FROM dist) as d1 JOIN (SELECT *,random() FROM  dist) d2 ON d1.a = d2.a JOIN (SELECT *,random() FROM dist) as d3 ON d3.a = d2.a;

 SELECT count(*) FROM (SELECT *,random() FROM dist) as d1 LEFT JOIN (SELECT *,random() FROM  dist) d2 ON d1.a = d2.a LEFT JOIN (SELECT *,random() FROM dist) as d3 ON d3.a = d2.a;

-- router query
SELECT count(*) FROM dist d1 LEFT JOIN dist d2 ON d1.a = d2.a LEFT JOIN dist d3 ON d3.a = d2.a WHERE d1.a = 1 AND d2.a = 1 AND d3.a = 1;

-- fast path router query
SELECT count(*) FROM dist WHERE a = 1;

-- simple INSERT
INSERT INTO dist (a) VALUES (1);

-- repartition join, probably not relevant, but still be defensive
 SELECT count(*) FROM dist d1 JOIN dist d2 ON d1.b = d2.b JOIN dist d3 ON d3.b = d2.b;

-- update query with join
UPDATE dist SET b = foo.a FROM (SELECT d1.a FROM dist d1 JOIN dist d2 USING(a))foo

-- insert select pushdown
INSERT INTO dist
 SELECT d1.* FROM dist d1 JOIN dist d2 ON d1.a = d2.a JOIN dist d3 ON d3.a = d2.a;

-- insert select via coordinator
INSERT INTO dist
 SELECT d1.* FROM dist d1 JOIN dist d2 ON d1.a = d2.a JOIN dist d3 ON d3.a = d2.a OFFSET 0;

-- insert select via repartitioning
 INSERT INTO dist (a)
 SELECT max(d1.b) FROM dist d1 JOIN dist d2 ON d1.a = d2.a JOIN dist d3 ON d3.a = d2.a GROUP BY d1.a;
```

Reply:
I think it'd even be better if we add a new test file: `geqo.sql`  and push the limits with setting the following at the top:
```SQL
SET geqo_threshold=2;
SET geqo_generations=1000;
SETT geqo_pool_size=1000;
```

Reply:
ping on this.


Also, can you make sure that the  tests pass and the style check is good?